tools.yml

```diff
@@ -605,7 +611,33 @@ tools:
     cores: 16
     mem: 64
   toolshed.g2.bx.psu.edu/repos/galaxy-australia/smudgeplot/smudgeplot/.*:
-    mem: 8
+    cores: 8
+    rules:
+    - id: tpvdb_smudgeplot_1_input_rule
+      if: input_size < 1
+      mem: 15
+    - id: tpvdb_smudgeplot_5_input_rule
+      if: 1 <= input_size < 5
+      mem: 75
+    - id: tpvdb_smudgeplot_10_input_rule
+      if: 5 <= input_size < 10
+      cores: 12
+      mem: 150
+    - id: tpvdb_smudgeplot_15_input_rule
+      if: 10 <= input_size < 15
+      cores: 12
+      mem: 225
+    - id: tpvdb_smudgeplot_20_input_rule
+      if: 15 <= input_size < 20
+      cores: 12
+      mem: 300
+    - id: tpvdb_smudgeplot_25_input_rule
```

Comment:
I wonder whether we can simplify these to Goldilocks style small, medium and large rules, and make the medium rule a calculation based on input_size? It looks like the medium rule would be something like:
```
    - id: tpvdb_smudgeplot_medium_input_rule
      if: 5 <= input_size < 20
      cores: 12
      mem: 75 + (input_size * 15)
```